spyder/plugins/editor/widgets/editor.py

```diff
@@ -1636,7 +1636,13 @@ def close_file(self, index=None, force=False):
 
             filename = self.data[index].filename
             self.remove_from_data(index)
-            finfo.editor.notify_close()
+            editor = finfo.editor
+            editor.notify_close()
+            editor.setParent(None)
+            editor.completion_widget.setParent(None)
+            if self.parent():
+                # Can be false in tests

```

Comment:
```suggestion
                # Can be None in tests
```

I think that's really the case.